NullPointerException if an exception is thrown from one way thrift method

If a thrift service throws an exception on one-way method, it causes NullPointerException because  ThriftFunction does not have a value if the method is one-way.

```java
@Mock
private OnewayHelloService.Iface serviceHandler = mock(OnewayHelloService.Iface.class);

@Rule
public final ServerRule server = new ServerRule() {
    @Override
    protected void configure(ServerBuilder sb) throws Exception {
        sb.service("/thrift", THttpService.of(serviceHandler));
    }
};

@Test
public void execute_oneway() throws Exception {
    OnewayHelloService.Iface client = new ClientBuilder(server.uri(BINARY, "/thrift"))
            .build(OnewayHelloService.Iface.class);
    doThrow(new IllegalArgumentException())
            .doThrow(new IllegalArgumentException())
            .doNothing()
            .when(onewayServiceHandler).hello(anyString());
    client.hello("hello");
    verify(onewayServiceHandler, times(3)).hello("hello");
}
```

```
20:52:32.367 [armeria-common-blocking-tasks-2-1] WARN  c.l.a.common.util.CompletionActions - Unexpected exception from a completion action:
java.util.concurrent.CompletionException: java.lang.NullPointerException
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:824)
	at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:797)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at com.linecorp.armeria.common.DefaultRpcResponse.completeExceptionally(DefaultRpcResponse.java:75)
	at com.linecorp.armeria.server.thrift.ThriftCallService.lambda$invokeSynchronously$0(ThriftCallService.java:176)
	at com.linecorp.armeria.common.AbstractRequestContext.lambda$makeContextAware$1(AbstractRequestContext.java:72)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException: null
	at com.linecorp.armeria.internal.thrift.ThriftFunction.newResult(ThriftFunction.java:219)
	at com.linecorp.armeria.server.thrift.THttpService.handleException(THttpService.java:661)
	at com.linecorp.armeria.server.thrift.THttpService.lambda$invoke$4(THttpService.java:595)
	at com.linecorp.armeria.common.util.Functions.lambda$voidFunction$1(Functions.java:192)
	at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:822)
	... 10 common frames omitted
```

--- a/thrift/src/main/java/com/linecorp/armeria/server/thrift/THttpService.java
+++ b/thrift/src/main/java/com/linecorp/armeria/server/thrift/THttpService.java
@@ -591,13 +591,13 @@ public final class THttpService extends AbstractHttpService {
         }
 
         reply.handle(voidFunction((result, cause) -> {
-            if (cause != null) {
-                handleException(ctx, reply, res, serializationFormat, seqId, func, cause);
+            if (func.isOneWay()) {
+                handleOneWaySuccess(ctx, reply, res, serializationFormat);
                 return;
             }
 
-            if (func.isOneWay()) {
-                handleOneWaySuccess(ctx, reply, res, serializationFormat);
+            if (cause != null) {
+                handleException(ctx, reply, res, serializationFormat, seqId, func, cause);
                 return;
             }
 

--- a/thrift/src/test/java/com/linecorp/armeria/client/thrift/ThriftOverHttpClientTest.java
+++ b/thrift/src/test/java/com/linecorp/armeria/client/thrift/ThriftOverHttpClientTest.java
@@ -110,11 +110,16 @@ public class ThriftOverHttpClientTest {
     private static final HelloService.AsyncIface exceptionThrowingHandler = (name, resultHandler)
             -> resultHandler.onError(new Exception(name));
 
+    private static final OnewayHelloService.AsyncIface exceptionThrowingOnewayHandler =
+            (name, resultHandler) -> {
+                assertThat(serverReceivedNames.add(name)).isTrue();
+                resultHandler.onError(new Exception(name));
+            };
+
     private static final OnewayHelloService.AsyncIface onewayHelloHandler = (name, resultHandler) -> {
         resultHandler.onComplete(null);
         assertThat(serverReceivedNames.add(name)).isTrue();
     };
-
     private static final DevNullService.AsyncIface devNullHandler = (value, resultHandler) -> {
         resultHandler.onComplete(null);
         assertThat(serverReceivedNames.add(value)).isTrue();
@@ -144,6 +149,8 @@ public class ThriftOverHttpClientTest {
         HELLO(helloHandler, HelloService.Iface.class, HelloService.AsyncIface.class),
         EXCEPTION(exceptionThrowingHandler, HelloService.Iface.class, HelloService.AsyncIface.class),
         ONEWAYHELLO(onewayHelloHandler, OnewayHelloService.Iface.class, OnewayHelloService.AsyncIface.class),
+        EXCEPTION_ONEWAY(exceptionThrowingOnewayHandler, OnewayHelloService.Iface.class,
+                         OnewayHelloService.AsyncIface.class),
         DEVNULL(devNullHandler, DevNullService.Iface.class, DevNullService.AsyncIface.class),
         BINARY(binaryHandler, BinaryService.Iface.class, BinaryService.AsyncIface.class),
         TIME(timeServiceHandler, TimeService.Iface.class, TimeService.AsyncIface.class),
@@ -362,6 +369,38 @@ public class ThriftOverHttpClientTest {
         OnewayHelloService.AsyncIface client =
                 Clients.newClient(clientFactory(), getURI(Handlers.ONEWAYHELLO),
                                   Handlers.ONEWAYHELLO.asyncIface(), clientOptions);
+        BlockingQueue<Object> resQueue = new LinkedBlockingQueue<>();
+
+        String[] names = { "kukuman", "kukuman2" };
+        for (String name : names) {
+            client.hello(name, new RequestQueuingCallback(resQueue));
+        }
+
+        for (String ignored : names) {
+            assertThat(resQueue.take()).isEqualTo("null");
+        }
+
+        for (String ignored : names) {
+            assertThat(serverReceivedNames.take()).isIn(names);
+        }
+    }
+
+    @Test(timeout = 10000)
+    public void testExceptionThrowingOnewayServiceSync() throws Exception {
+        OnewayHelloService.Iface client =
+                Clients.newClient(clientFactory(), getURI(Handlers.EXCEPTION_ONEWAY),
+                                  Handlers.EXCEPTION_ONEWAY.iface(), clientOptions);
+        client.hello("kukuman");
+        client.hello("kukuman2");
+        assertThat(serverReceivedNames.take()).isEqualTo("kukuman");
+        assertThat(serverReceivedNames.take()).isEqualTo("kukuman2");
+    }
+
+    @Test(timeout = 10000)
+    public void testExceptionThrowingOnewayServiceAsync() throws Exception {
+        OnewayHelloService.AsyncIface client =
+                Clients.newClient(clientFactory(), getURI(Handlers.EXCEPTION_ONEWAY),
+                                  Handlers.EXCEPTION_ONEWAY.asyncIface(), clientOptions);
         BlockingQueue<Object> resQueue = new LinkedBlockingQueue<>();
 
         String[] names = { "kukuman", "kukuman2" };

